doc/man/man3/crypto_x25519_dangerous_fast.3monocypher

```diff
@@ -0,0 +1,135 @@
+.\" worldwide.  This software is distributed without any warranty.
+.\"
+.\" You should have received a copy of the CC0 Public Domain Dedication along
+.\" with this software.  If not, see
+.\" <https://creativecommons.org/publicdomain/zero/1.0/>
+.\"
+.Dd March 24, 2020
+.Dt CRYPTO_X25519_DANGEROUS_FAST 3monocypher
+.Os
+.Sh NAME
+.Nm crypto_x25519_dangerous_fast ,
+.Nm crypto_x25519_dangerous_small
+.Nd generation of Curve25519 points including low-order points
+.Sh SYNOPSIS
+.In monocypher.h
+.Ft void
+.Fo crypto_hidden_to_curve_fast
+.Fa "uint8_t pk[32]"
+.Fa "const uint8_t sk[32]"
+.Fc
+.Ft void
+.Fo crypto_hidden_to_curve_small
+.Fa "uint8_t pk[32]"
+.Fa "const uint8_t sk[32]"
+.Fc
+.Sh DESCRIPTION
+These functions are used in public key generation for
+.Xr crypto_curve_to_hidden 3monocypher .
+.Sy This is a highly advanced feature ;
+unless you are reading this because you were referred here from
+.Xr crypto_curve_to_hidden 3monocypher ,
+.Sy you likely have no reason to be using these functions
+and are probably looking for
+.Xr crypto_key_exchange 3monocypher
+or
+.Xr crypto_x25519 3monocypher
+instead.
+Expect elliptic curve jargon on this page.
+.Pp
+Both functions generate a Curve25519 public key
+.Fa pk
+from the given secret key
+.Fa sk ;
+the public keys are on the
+.Em whole
+curve, rather than just the main prime-order subgroup.
+but do the same with different code size and memory characteristics:
+.Fn crypto_x25519_dangerous_fast
+uses multiple large temporary variables and uses functions that are
+normally used internally for
+.Xr crypto_sign 3monocypher ;
+accordingly, it uses both more memory (for the temporary variables) and
+more code size (unless the signing code is already compiled in
+elsewhere).
+.Fn crypto_x25519_dangerous_small
+yields the same result, but does so using less time and memory at a
+large performance penalty compared to
+.Fn crypto_x25519_dangerous_fast .
+.Pp
+The resulting public keys
+.Em may
+be used with
+.Xr crypto_x25519 3monocypher
+and
+.Xr crypto_key_exchange 3monocypher ,
+which clear the cofactor,
+but are dangerous for other uses because they contain low-order points,
+and may trip up other implementations of X25519.
+.Sh RETURN VALUES
+These functions have no return value.
+They cannot fail.
+.Sh SEE ALSO
+.Xr crypto_curve_to_hidden 3monocypher ,
+.Xr crypto_key_exchange 3monocypher ,
+.Xr crypto_x25519 3monocypher ,
+.Xr intro 3monocypher
+.Sh HISTORY
+The
+.Fn crypto_x25519_dangerous_fast
+and
+.Fn crypto_x25519_dangerous_small
+functions first appeared in Monocypher 3.1.0.
+.Sh IMPLEMENTATION DETAILS
+The slow variant is approximately an entire 1.8 times slower than the
+fast variant.
+When considering that, on average, two calls to this function will be
+required for obtaining a valid keypair for
+.Xr crypto_curve_to_hidden 3monocypher ,
+this adds up to an
+.Em average
+effective slowdown for key pair generation of a factor of 3.6.
```

Comment:
Neat trick: we can speed up *clean* key generation by using the fast "dangerous" version with a pre-clamped scalar (`scalar[0] &= 0xf8`). Not sure we want to document this, but users wanting to speed up ephemeral key pair generation can now use this.

Or we could add a dedicated function…